src/sync/impl/sync_file.hpp

```diff
@@ -83,10 +83,10 @@ class SyncFileManager {
     static bool try_file_exists(const std::string& path) noexcept;
 
     /// Return the path for a given Realm, creating the user directory if it does not already exist.
-    std::string realm_file_path(const std::string& local_user_identity, const std::string& realm_file_name) const;
+    std::string realm_file_path(const std::string& user_identity, const std::string& local_user_identity, const std::string& realm_file_name) const;
```

Comment:
Would it be cleaner to send in the `User` object instead of having to read two strings from it?

Reply:
SyncFileManager isn't bound to User, but it is SyncManager handles Users. 

I believe the reason behind this decision is to facilitate testing SyncFileManager as there is no straight way to mock Users.